adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/VertxBasedRestProtocolAdapter.java

```diff
@@ -12,39 +12,52 @@
 
 package org.eclipse.hono.adapter.rest;
 
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BasicAuthHandler;
 import org.eclipse.hono.adapter.http.AbstractVertxBasedHttpProtocolAdapter;
-import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.adapter.rest.credentials.BasicAuthProvider;
 
 import io.vertx.core.http.HttpMethod;
 import io.vertx.ext.web.Router;
-import io.vertx.ext.web.RoutingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A Vert.x based Hono protocol adapter for accessing Hono's Telemetry &amp; Event API using REST.
  */
-public final class VertxBasedRestProtocolAdapter extends AbstractVertxBasedHttpProtocolAdapter<ServiceConfigProperties> {
+public class VertxBasedRestProtocolAdapter
+        extends AbstractVertxBasedHttpProtocolAdapter<RestProtocolAdapterProperties> {
 
-    private static final String PARAM_TENANT = "tenant";
+    private static final Logger LOG             = LoggerFactory.getLogger(VertxBasedRestProtocolAdapter.class);
+    private static final String PARAM_TENANT    = "tenant";
     private static final String PARAM_DEVICE_ID = "device_id";
 
     @Override
     protected void addRoutes(final Router router) {
+        if (getConfig().isAuthenticationRequired()) {
```

Comment:
As commented already in previous PR: 
please change to WARN if the authentication is DISABLED, and do not log anything if it is enabled (which is the default anyway).